### ISSUE TITLE
No /proc mount

### ISSUE THREAD

**Issue by deliciouslytyped** [t184256/nix-on-droid#77]
Stuff like `ps` doesn't work, I'm assuming because /proc isn't mounted,
am I missing something?

**Comment by t184256** [comment]
Huh? I'm not doing anything special to `/proc` it should be the same as for the 'parent' Android system. `ls /proc`.

We probably need to pick some patches from termux to work around Android restrictions, the way @Gerschtli  [did that](https://github.com/t184256/nix-on-droid/blob/master/overlays/htop.nix) for `htop`.

**Comment by Gerschtli** [comment]
/proc is mounted but mainly not accessible without a rooted phone.

If you try `ls -l /proc` you will see, what I mean.

**Comment by deliciouslytyped** [comment]
Sorry, I wasn't at all thorough with the original post because I was distracted.
The real issue is calling ps on termux will yield a result, whereas on nix-on-droid:
```
Error: /proc must be mounted
  To mount /proc at boot you need an /etc/fstab line like:
      proc   /proc   proc    defaults
  In the meantime, run "mount proc /proc -t proc"
```

**Comment by Gerschtli** [comment]
Yes, that is because nix-on-droid does not have read access to most part of /proc. termux does have the same problem where it uses custom patches to make some programs work.

**Comment by deliciouslytyped** [comment]
Hmm. So should one start ingesting patches from termux into a package set?

**Comment by Gerschtli** [comment]
That would be a valid approach very much like I did it with the htop overlay. So if you want to contribute, just add an overlay in the overlays directory and add in overlays/default.nix. It would be awesome to see ps and others fixed in nix-on-droid! :)

**Comment by deliciouslytyped** [comment]
On the bucket list it goes. :P

**Comment by lypanov** [comment]
Looks like it's possible to get this working without overlay hacks. Wouldn't this workaround also work for nix-on-droid?

https://github.com/sdrausty/TermuxArch/issues/175
